app/src/main/java/com/github/sdpteam15/polyevents/view/activity/TimeTableActivity.kt

```diff
@@ -0,0 +1,510 @@
+import android.os.Handler
+import android.util.TypedValue
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.cardview.widget.CardView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import com.github.sdpteam15.polyevents.R
+import com.github.sdpteam15.polyevents.helper.HelperFunctions
+import com.github.sdpteam15.polyevents.model.database.remote.Database
+import com.github.sdpteam15.polyevents.model.entity.Event
+import com.github.sdpteam15.polyevents.model.observable.ObservableList
+import com.github.sdpteam15.polyevents.model.observable.ObservableMap
+import com.github.sdpteam15.polyevents.view.fragments.EXTRA_EVENT_ID
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.collections.ArrayList
+
+/**
+ * Activity displaying the timetable of the event
+ */
+class TimeTableActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {
+
+    companion object{
+        var instance: TimeTableActivity? = null
+    }
+
+    private lateinit var leftButton: ImageButton
+    private lateinit var rightButton: ImageButton
+    private lateinit var spinner: Spinner
+
+    private var obsEventsMap = ObservableMap<String, ObservableMap<String, ObservableList<Event>>>()
+    private var obsZoneNames = ObservableMap<String, String>()
+    var selectedItem = 0
+
+    val displayedViews = ArrayList<View>()
+    private val idViewToIdEvent = mutableMapOf<Int, String>()
+
+    var nextId = 156
+    val hourSizeDp = 100
+    val lineHeightDp = 2
+    var currentPadding = 30
+    val linepaddingLeftDP = 2
+    val nowLineHeightDP = 2
+
+    //Event params
+    var widthDP = 250
+
+    private var idNowBar: View? = null
+    private var selectedDate = LocalDateTime.now()
+    private var selectedZone: String? = null
+
+
+    val hourToLine: MutableMap<Int, Int> = mutableMapOf()
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_time_table)
+        instance = this
+        leftButton = findViewById(R.id.id_change_zone_left)
+        rightButton = findViewById(R.id.id_change_zone_right)
+
+        //Setup of buttons
+        rightButton.setOnClickListener {
+            selectedDate = selectedDate.plusDays(1)
+            setupDate()
+            drawZoneEvents(selectedZone!!)
+        }
+        leftButton.setOnClickListener {
+            selectedDate = selectedDate.minusDays(1)
+            setupDate()
+            drawZoneEvents(selectedZone!!)
+        }
+        spinner = findViewById(R.id.id_title_zone)
+
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        setupSpinner()
+        setupDate()
+
+        generateTime(0, 23)
+        getEventsFromDB()
+        addNowBar()
+    }
+
```

Comment:
to avoid leaks on `instance`, it would be necessary to set it to null in onDestroy of the activity
```suggestion

    override fun onDestroy() {
        super.onDestroy()
        instance = null
    }
    
```

Reply:
Yes, that's very pertinent